lib/python/rose/bush.py

```diff
@@ -527,28 +549,46 @@ def view(self, user, suite, path, path_in_tar=None, mode=None):
             file_content = "rose-conf"
         else:
             file_content = self.suite_engine_proc.is_conf(path)
-        template = self.template_env.get_template("view.html")
+
+        if search_string:
+            # client is performing a search
+            template = self.template_env.get_template("file_view.html")
+            results = []
+            line_numbers = []
+
+            # perform search
+            for i, line in enumerate(lines):
+                if search_mode == SEARCH_MODE_REGEX:
+                    match = re.search(search_string, line)
+                    if not match:
+                        continue
+                    start, end = match.span()
+                elif search_mode == SEARCH_MODE_TEXT:
+                    match = line.find(search_string)
+                    if match == -1:
+                        continue
+                    start = match
+                    end = start + len(search_string)
+                else:
+                    # ERROR: un-reccognised search_mode
+                    pass
+                # if line matches search string include in results
+                results.append([line[:start], line[start:end],
+                                line[end:]])
+                line_numbers.append(i + 1) # line numbers start from 1
+            lines = results
+        else:
+            # no search is being performed, client is requesting the whole
+            # page
+            line_numbers = range(1, len(lines) + 1)
+            lines = [[line] for line in lines]
 
         data = {}
         data.update(self._get_suite_logs_info(user, suite))
+        template = self.template_env.get_template("file_view.html")
```

Comment:
The other files in the directory use hyphen (`-`) instead of underscore (`_`) as separator.